tests/integration/istio_test.go

```diff
@@ -0,0 +1,496 @@
+
+		deleteAllIstiosAndRevisions(ctx)
+	})
+
+	Describe("basic operation", func() {
+		BeforeAll(func() {
+			Step("Creating the custom resource")
+			istio = &v1alpha1.Istio{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: istioName,
+				},
+				Spec: v1alpha1.IstioSpec{
+					Version:   istioVersion,
+					Namespace: istioNamespace,
+					UpdateStrategy: &v1alpha1.IstioUpdateStrategy{
+						Type: v1alpha1.UpdateStrategyTypeInPlace,
+						InactiveRevisionDeletionGracePeriodSeconds: ptr.Of(int64(gracePeriod.Seconds())),
+					},
+					Values: []byte(`{
+						"pilot":{"image":"` + pilotImage + `"},
+						"istio_cni":{"enabled":true}
+					}`),
+				},
+			}
+			Expect(k8sClient.Create(ctx, istio)).To(Succeed())
+		})
+
+		It("updates the Istio resource", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, istioKey, istio)).To(Succeed())
+				g.Expect(istio.Status.ObservedGeneration).To(Equal(istio.Generation))
+			}).Should(Succeed())
+		})
+
+		It("creates the IstioRevision resource", func() {
+			revKey := client.ObjectKey{Name: istioName}
+			rev := &v1alpha1.IstioRevision{}
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, revKey, rev)).To(Succeed())
+				g.Expect(rev.GetOwnerReferences()).To(ContainElement(NewOwnerReference(istio)))
+			}).Should(Succeed())
+
+			Expect(rev.Spec).To(Equal(v1alpha1.IstioRevisionSpec{
+				Version:   istio.Spec.Version,
+				Namespace: istio.Spec.Namespace,
+				Values: deprettify([]byte(`{
+					"defaultRevision":"",
+					"gateways":{
+						"istio-egressgateway":{},
+						"istio-ingressgateway":{}
+					},
+					"global":{
+						"configValidation":true,
+						"istioNamespace":"` + istio.Spec.Namespace + `"
+					},
+					"istio_cni": {
+						"enabled":true
+					},
+					"pilot":{
+						"image":"maistra.io/test:latest"
+					},
+					"revision":"` + revKey.Name + `"
+				}`)),
+			}))
+		})
+
+		When("the underlying IstioRevision is deleted", func() {
+			rev := &v1alpha1.IstioRevision{}
+			revKey := client.ObjectKey{Name: istioName}
+
+			BeforeAll(func() {
+				rev = &v1alpha1.IstioRevision{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: istio.Name,
+					},
+				}
+				Expect(k8sClient.Delete(ctx, rev)).To(Succeed())
+			})
+
+			It("recreates the IstioRevision", func() {
+				Eventually(k8sClient.Get).WithArguments(ctx, revKey, rev).Should(Succeed())
+				Expect(rev.GetOwnerReferences()).To(ContainElement(NewOwnerReference(istio)))
+				Expect(rev.Spec).To(Equal(v1alpha1.IstioRevisionSpec{
+					Version:   istio.Spec.Version,
+					Namespace: istio.Spec.Namespace,
+					Values: deprettify([]byte(`{
+						"defaultRevision":"",
+						"gateways":{
+							"istio-egressgateway":{},
+							"istio-ingressgateway":{}
+						},
+						"global":{
+							"configValidation":true,
+							"istioNamespace":"` + istio.Spec.Namespace + `"
+						},
+						"istio_cni": {
+							"enabled":true
+						},
+						"pilot":{
+							"image":"maistra.io/test:latest"
```

Comment:
same here, use `pilotImage` instead of duplicating the image name

Reply:
done